### ISSUE TITLE
ssh jail not recognizing fails when recidive is used or when log level set to 4

### ISSUE THREAD

**Issue by aptalca** [fail2ban/fail2ban#1340]
### Environment:

- Fail2Ban version (including any possible distribution suffixes): 0.8.11
- OS, including release name/version: Ubuntu 14.04.4 (GNU/Linux 3.13.0-71-generic x86_64)
- [x] Fail2Ban installed via OS/distribution mechanisms
- [x] You have not applied any additional foreign patches to the codebase
- [ ] Some customizations were done to the configuration (provide details below is so)

### The issue:

If ssh is the only enabled jail (with log level set to 3), it works just fine. It recognizes fails and bans offenders. 

But if I also enable recidive, ssh no longer recognizes fails. Interestingly, even if recidive is disabled, when log level is set to 4, ssh is not recognizing fails either. With log level set to 3 and recidive is enabled, there is nothing in the logs that suggest an error. ssh jail seems to start and is listed as active in status.

This same configuration used to work fine with ssh and recidive up until the latest kernel update to 14.04.4 (3.13.0-71). After the update, ssh no longer works when recidive is active.

#### Steps to reproduce

Update to 14.04.4 (3.13.0-71), enable both ssh and recidive jails, wait for an offender to try and connect to the ssh port with a nonexistent user.

#### Expected behavior

For ssh jail to recognize a failed attempt

#### Observed behavior

No failed attempts recognized

#### Any additional information

To test further, I tried when I had recent offenders in the log. When I enable just the ssh jail, as soon as fail2ban is started, ssh filter recognizes the multiple failed attempts and bans the ip. If I enable recidive also and restart fail2ban, ssh jail doesn't recognize any failed attempts. If I disable recidive and restart, ssh jail again recognizes the failed attempts and bans the ip immediately.

### Configuration, dump and another helpful excerpts

There are no errors in the logs when level set to 4, even when the ssh jail is not recognizing any failed attempts.

#### Any customizations done to /etc/fail2ban/ configuration

None. To confirm, I reinstalled fail2ban from scratch and only added a jail.local to enable the ssh and recidive jails

#### Relevant parts of /var/log/fail2ban.log file:

No errors

#### Relevant lines from monitored log files in question:

The following lines repeat many many times when log level set to 4 and both recidive and ssh jails are enabled:
```
2016-02-29 12:58:01,777 fail2ban.filter.datedetector: DEBUG  Got time using template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,777 fail2ban.filter.datedetector: DEBUG  Matched time template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,778 fail2ban.filter.datedetector: DEBUG  Got time using template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,778 fail2ban.filter.datedetector: DEBUG  Matched time template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,778 fail2ban.filter.datedetector: DEBUG  Got time using template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,778 fail2ban.filter.datedetector: DEBUG  Matched time template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,778 fail2ban.filter.datedetector: DEBUG  Got time using template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,778 fail2ban.filter.datedetector: DEBUG  Matched time template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,778 fail2ban.filter.datedetector: DEBUG  Got time using template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,779 fail2ban.filter.datedetector: DEBUG  Matched time template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,779 fail2ban.filter.datedetector: DEBUG  Got time using template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,779 fail2ban.filter.datedetector: DEBUG  Matched time template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,779 fail2ban.filter.datedetector: DEBUG  Got time using template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,779 fail2ban.filter.datedetector: DEBUG  Matched time template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,779 fail2ban.filter.datedetector: DEBUG  Got time using template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,779 fail2ban.filter.datedetector: DEBUG  Matched time template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,780 fail2ban.filter.datedetector: DEBUG  Got time using template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,780 fail2ban.filter.datedetector: DEBUG  Matched time template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,780 fail2ban.filter.datedetector: DEBUG  Got time using template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,780 fail2ban.filter.datedetector: DEBUG  Matched time template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,780 fail2ban.filter.datedetector: DEBUG  Got time using template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,780 fail2ban.filter.datedetector: DEBUG  Matched time template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,780 fail2ban.filter.datedetector: DEBUG  Got time using template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,780 fail2ban.filter.datedetector: DEBUG  Matched time template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,781 fail2ban.filter.datedetector: DEBUG  Got time using template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,781 fail2ban.filter.datedetector: DEBUG  Matched time template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,781 fail2ban.filter.datedetector: DEBUG  Got time using template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,781 fail2ban.filter.datedetector: DEBUG  Matched time template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,781 fail2ban.filter.datedetector: DEBUG  Got time using template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,781 fail2ban.filter.datedetector: DEBUG  Matched time template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,781 fail2ban.filter.datedetector: DEBUG  Got time using template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,781 fail2ban.filter.datedetector: DEBUG  Matched time template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,782 fail2ban.filter.datedetector: DEBUG  Got time using template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,782 fail2ban.filter.datedetector: DEBUG  Matched time template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,782 fail2ban.filter.datedetector: DEBUG  Got time using template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,782 fail2ban.filter.datedetector: DEBUG  Matched time template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,782 fail2ban.filter.datedetector: DEBUG  Got time using template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,782 fail2ban.filter.datedetector: DEBUG  Matched time template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,782 fail2ban.filter.datedetector: DEBUG  Got time using template Year-Month-Day Hour:Minute:Second[,subsecond]
2016-02-29 12:58:01,783 fail2ban.filter.datedetector: DEBUG  Matched time template Year-Month-Day Hour:Minute:Second[,subsecond]
```

**Comment by aptalca** [comment]
I have been testing more jails and it is confusing. This time I tried just the roundcube-auth. It was working fine yesterday along with ssh and other jails, but today, I can't get it to work with just roundcube-auth enabled. I tried to set the log level to 4. Failed attempts are logged in the roundcube log. The fail2ban log shows the datedetector detecting each line from the log, but it is not recognizing any failed attempts.

There has to be something else going on that is causing these intermittent issues.

**Comment by sebres** [comment]
Enabling recidive together with detailed logging (like DEBUG) is very very bad idea, because recidive jail (filter) curently works using parse of fail2ban.log, so it will produce changes and scan for changes again and again (endless).

See also https://github.com/fail2ban/fail2ban/blob/master/config/jail.conf#L734

In addition, the version 0.8 is too old (I know that the maintainers of many distributions still use that, but it is old anyway).

**Comment by aptalca** [comment]
I realize that recidive is recommended against being used with debug logging, but that was not the issue.

Even with recidive disabled, ssh jail fails with debug logging on. I only turned debug logging on to see if there were any errors that would show why ssh jail is failing to catch anything.

If you closed this issue because 0.8 is old, I kind of understand. But does that mean that 0.8.11 is no longer supported even though it is still the official package version for ubuntu 12.04? Fail2ban.org still lists 0.8.14 as a "very stable" version under latest versions.

Is there a recommended way to update to 0.9.3 on ubuntu 12.04?

Thanks

**Comment by sebres** [comment]
> but that was not the issue.
> Even with recidive disabled, ssh jail fails with debug logging on

Then why you've mixed it in this issue with recidive jail? 
Then I miss the log entries, you mean will not be recognized.

But I can imagine that the entries that are written in fail2ban.log, interfere with the detection of the sshd filter, for example with systemd backend (no clue whether 0.8 this backend supported). But quite other scenarios are as well conceivably. 
The log excerpt that you've enclosed to the issue, suggests that recidive was on, and as already said it's evil.

> If you closed this issue because 0.8 is old

Indeed 0.8 is too old, and I've no idea why the maintainers many distribution does not actualize it to newer version. The fact is 0.8 will be no longer fixed.

> Is there a recommended way to update to 0.9.3 on ubuntu 12.04?

Well, the prerequirements are well known:
- Python >=2.6 or Python >=3.2 or PyPy.

The decision is up to you.
